service-impl/cache-service-impl/etcd-backing-store/src/main/java/uk/gov/gchq/palisade/cache/service/impl/EtcdBackingStore.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2018 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.cache.service.impl;
+
+import com.coreos.jetcd.Client;
+import com.coreos.jetcd.KV;
+import com.coreos.jetcd.Lease;
+import com.coreos.jetcd.data.ByteSequence;
+import com.coreos.jetcd.data.KeyValue;
+import com.coreos.jetcd.kv.GetResponse;
+import com.coreos.jetcd.kv.PutResponse;
+import com.coreos.jetcd.options.GetOption;
+import com.coreos.jetcd.options.PutOption;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+
+public class EtcdBackingStore implements BackingStore {
+
+    private Collection<String> connectionDetails;
+    private Client etcdClient;
+    private KV keyValueClient;
+    private Lease leaseClient;
+
+    public EtcdBackingStore() {
+    }
+
+    @JsonCreator
+    public EtcdBackingStore etcdClient(final Collection<String> connectionDetails) {
+        this.connectionDetails = connectionDetails;
+        this.etcdClient = Client.builder().endpoints(connectionDetails).build();
```

Comment:
please ensure previous connection (if any) is closed before replacing with new one